internal/source/http.go

```diff
@@ -0,0 +1,86 @@
+package source
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/nlepage/go-tarfs"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+)
+
+// http is a bundle source that sources bundles from the specified url.
+type HTTP struct {
+	client.Reader
+	SecretNamespace string
+}
+
+// Unpack unpacks a bundle by requesting the bundle contents from a specified URL
+func (b *HTTP) Unpack(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (*Result, error) {
+	if bundle.Spec.Source.Type != rukpakv1alpha1.SourceTypeHTTP {
+		return nil, fmt.Errorf("cannot unpack source type %q with %q unpacker", bundle.Spec.Source.Type, rukpakv1alpha1.SourceTypeHTTP)
+	}
+
+	url := bundle.Spec.Source.HTTP.URL
+	action := fmt.Sprintf("%s %s", http.MethodGet, url)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("create http request %q for bundle content: %v", action, err)
+	}
+	var userName, password string
+	if bundle.Spec.Source.HTTP.Auth.Secret.Name != "" {
+		userName, password, err = b.getCredentials(ctx, bundle)
+		if err != nil {
+			return nil, err
+		}
+		req.SetBasicAuth(userName, password)
+	}
+
+	httpClient := http.Client{Timeout: 10 * time.Second}
+	if bundle.Spec.Source.HTTP.Auth.InsecureSkipVerify {
+		tr := &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
```

Comment:
I'm not sure how much it matters, but the `http` package has a `DefaultTranport` that sets a lot more fields on this struct. Because of that, I did this in `rukpakctl run`:
```go
transport := http.DefaultTransport.(*http.Transport).Clone()
```

Reply:
Thanks!  I'll do the same way.